services/explorer/src/hooks/useExplorerNetwork.tsx

```diff
@@ -41,34 +34,34 @@ const initialState: Context = {
 const ExplorerNetworkContext = createContext<Context>(initialState);
 
 const useNetworkSelection = () => {
-  const networks = Object.values(Network);
+  const networks: string[] = Object.values(NetworkName);
   const hostNetwork = getDomainName();
-  const [currentNetwork, setCurrentNetwork] = useState<string>(
-    hostNetwork || Network.MAINNET
+  const [currentNetwork, setCurrentNetwork] = useState<NetworkName>(
+    hostNetwork || NetworkName.MAINNET
   );
-
+  const selectedNetwork = getNetwork(currentNetwork);
   const [graphQLClient, setGraphQLClient] = useState<Client>(
-    createClient(recreateGraphQLConnection(currentNetwork))
+    createClient(selectedNetwork.graphqlUrl)
   );
   const [backendEndpoint, setBackendEndpoint] = useState<string>(
-    recreateAPIConnection(currentNetwork)
+    selectedNetwork.indexerUrl
   );
   const [rpcEndpoint, setRpcEndpoint] = useState<string>(
-    recreateRPCConnection(currentNetwork)
+    selectedNetwork.rpcUrl
   );
   const [faucetEndpoint, setFaucetEndpoint] = useState<string>(
-    recreateFaucetConnection(currentNetwork)
+    selectedNetwork.faucetUrl
   );
 
   const setNetwork = (network: string) => {
-    setCurrentNetwork(network);
+    setCurrentNetwork(network as NetworkName);
   };
 
   useEffect(() => {
-    setGraphQLClient(createClient(recreateGraphQLConnection(currentNetwork)));
-    setBackendEndpoint(recreateAPIConnection(currentNetwork));
-    setRpcEndpoint(recreateRPCConnection(currentNetwork));
-    setFaucetEndpoint(recreateFaucetConnection(currentNetwork));
+    setGraphQLClient(createClient(getNetwork(currentNetwork).graphqlUrl));
+    setBackendEndpoint(getNetwork(currentNetwork).indexerUrl);
+    setRpcEndpoint(getNetwork(currentNetwork).rpcUrl);
+    setFaucetEndpoint(getNetwork(currentNetwork).faucetUrl);

```

Comment:
Can't we use `selectedNetwork` here too as it will change according to what is in the state either way?

Reply:
I missed that part. Yes we can use it there as well.